src/reggae/dub/interop/dublib.d

```diff
@@ -55,33 +55,52 @@ struct Dub {
         return _project.packageManager.getPackage(dubPackage, Version(version_));
     }
 
-    DubConfigurations getConfigs(in from!"reggae.options".Options options) {
+    static auto getGeneratorSettings(in Options options) {
+        import dub.compilers.compiler: getCompiler;
+        import dub.generators.generator: GeneratorSettings;
+        import std.path: baseName, stripExtension;
+
+        const compilerBinName = options.dCompiler.baseName.stripExtension;
+
+        GeneratorSettings ret;
+
+        ret.compiler = () @trusted { return getCompiler(compilerBinName); }();
+        ret.platform = () @trusted {
+            return ret.compiler.determinePlatform(ret.buildSettings,
+                options.dCompiler, options.dubArchOverride);
+        }();
+        ret.buildType = options.dubBuildType;
+
+        return ret;
+    }
+
+    DubConfigurations getConfigs(/*in*/ ref from!"dub.platform".BuildPlatform platform) {
 
         import std.algorithm.iteration: filter, map;
         import std.array: array;
 
-        auto settings = generatorSettings(options.dCompiler.toCompiler);
-
         // A violation of the Law of Demeter caused by a dub bug.
         // Otherwise _project.configurations would do, but it fails for one
         // projet and no reduced test case was found.
         auto configurations = _project
             .rootPackage
             .recipe
             .configurations
-            .filter!(c => c.matchesPlatform(settings.platform))
+            .filter!(c => c.matchesPlatform(platform))
             .map!(c => c.name)
             .array;
 
-        return DubConfigurations(configurations, _project.getDefaultConfiguration(settings.platform));
+        // Project.getDefaultConfiguration() requires a mutable arg (forgotten `in`)
+        return DubConfigurations(configurations, _project.getDefaultConfiguration(platform));
     }
 
     DubInfo configToDubInfo
-    (in from!"reggae.options".Options options, in string config)
+    (from!"dub.generators.generator".GeneratorSettings settings, in string config)
```

Comment:
Why is `settings` not `in`?

Reply:
Because of an ugly dub API design - it expects a whole new `GeneratorSettings` struct and doesn't take a `config` string. So we need a copy anyway to set `settings.config` and pass that to dub.